internal/pkg/aws/route53/route53.go

```diff
@@ -20,47 +23,130 @@ const (
 )
 
 type api interface {
-	ListHostedZonesByName(in *route53.ListHostedZonesByNameInput) (*route53.ListHostedZonesByNameOutput, error)
+	ListHostedZonesByName(*route53.ListHostedZonesByNameInput) (*route53.ListHostedZonesByNameOutput, error)
+	ListResourceRecordSets(*route53.ListResourceRecordSetsInput) (*route53.ListResourceRecordSetsOutput, error)
+}
+
+type nameserverResolver interface {
+	LookupNS(ctx context.Context, name string) ([]*net.NS, error)
 }
 
 // Route53 wraps an Route53 client.
 type Route53 struct {
 	client api
+	dns    nameserverResolver
+
+	hostedZoneIDFor map[string]string
```

Comment:
This is an interesting naming convention. What situations does `...For` apply to?